clientv3/balancer.go

```diff
@@ -126,11 +126,21 @@ func (b *simpleBalancer) updateAddrs(eps []string) {
 	b.addrs = addrs
 	// updating notifyCh can trigger new connections,
 	// but balancer only expects new connections if all connections are down
-	if b.pinAddr == "" {
+	// or updated addresses doesn't include the pin addr.
+	if b.pinAddr == "" || !b.hasPinAddr(addrs) {
```

Comment:
this needs to be redesigned; if `b.pinAddr != ""` then this can trigger the failure fixed by 0c4e67c1f445d9c3d6e3799f78301c682c8a57cf